src/pybind/mgr/cephadm/module.py

```diff
@@ -65,6 +65,11 @@
 HOST_CACHE_PREFIX = "host."
 SPEC_STORE_PREFIX = "spec."
 
+# ceph daemon types that use the ceph container image.
+# NOTE: listed in upgrade order!
+CEPH_TYPES = ['mgr', 'mon', 'crash', 'osd', 'mds', 'rgw', 'rbd-mirror']
```

Comment:
feels similar to the reconfig deps  (`_calc_daemon_deps`) ... should we make this more explicit by adding `_calc_upgrade_deps` func or similar?

Reply:
```
        need = {
            'prometheus': ['mgr', 'alertmanager', 'node-exporter'],
            'grafana': ['prometheus'],
            'alertmanager': ['alertmanager'],
        }
```
not sure these are related.. although it might make sense to move that constant to the top of the file too

Reply:
think I'm more worried about the name `CEPH_TYPES` being too vague and that the comment restricting this to an upgrade order would be overlooked ...

Reply:
``CEPH_TYPES_INORDER``?